handbook/github.md

```diff
@@ -0,0 +1,99 @@
+
+## Step 4: Do the work
+
+Once you've decided what you are going to do, and you have shared your plans with the team, it's time to start working.  You should make sure that you have an up-to-date fork of the repository you plan to work in, and that you have cloned and pulled down the most recent changes of this fork to your local workspace.  You should make sure that you have an up-to-date environment for the work (e.g., conda), and that you have all the appropriate tools needed for the specific repository, including testing tools (e.g., pytest), linting tools (e.g., pre-commit), etc.  Verify that the testing and development tools in your environment work before starting to do development.
+
+Follow the normal git best practices, and push changes up to your fork only when you know that things are working.  If you are adding lines of code, make sure those lines of code are being tested!  If you are deleting lines of code, make sure those deletions aren't changing any existing tests (or, if appropriate, change the tests accordingly).  If you've added a feature, make sure that some form of documentation (e.g., API documentation) is being added to the package.
+
+:::{admonition} Create Draft Pull Requests early!
+:class: tip
+You can create a Pull Request for new work in your fork *at any time.*  If you want, you can create the Pull Request only when you are pretty confident that you are close or ready to merge into the main branch.  However, we recommend creating your Pull Requests in *draft mode* at an early stage, possibly even after your very first commit to your fork's branch.  Sharing your Pull Requests with the rest of the team helps other people follow along with what Xdev is doing!  And it helps us learn from each other *while* we are doing our work.
+:::
+
+## Step 5: Merge
+
+Once you thing you have a Pull Request that is ready to merge, convert your Pull Request to from a *draft* to a regular Pull Request.  If this does not automatically trigger reviews on your Pull Request from the team, then select a couple people to review your work.  Try not to alway select the same people, though!  It's good to share the review workload around.
+
+At this point, someone on the team should review your Pull Request.  If nobody is reviewing your Pull Request after a day of waiting, give the selected reviewers a gently nudge with a *direct mention* in the Pull Request conversation.  If the selected reviewer is just too swamped to get to it within 24 hours of receiving the notification, then the reviewer should suggest another reviewer for the Pull Request.  If the selected reviewer has time to conduct the review, but it might take them a while, then the reviewer should let the author of the Pull Request know that they are looking at it and that it might take a little bit.
```

Comment:
```suggestion
At this point, someone on the team should review your Pull Request.  If nobody is reviewing your Pull Request after a day of waiting, give the selected reviewers a gentle nudge with a *direct mention* in the Pull Request conversation.  If the selected reviewer is just too swamped to get to it within 24 hours of receiving the notification, then the reviewer should suggest another reviewer for the Pull Request.  If the selected reviewer has time to conduct the review, but it might take them a while, then the reviewer should let the author of the Pull Request know that they are looking at it and that it might take a little bit.
```